core/prover/src/server.rs

```diff
@@ -0,0 +1,193 @@
+mod pool;
+
+// Built-in
+use std::sync::{Arc, RwLock};
+use std::thread;
+use std::{net, time};
+// External
+use actix_web::{web, App, HttpResponse, HttpServer};
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+// Workspace deps
+
+struct AppState {
+    connection_pool: storage::ConnectionPool,
+    preparing_data_pool: Arc<RwLock<pool::ProversDataPool>>,
+    prover_timeout: time::Duration,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ProverReq {
+    pub name: String,
+}
+
+fn register(data: web::Data<AppState>, r: web::Json<ProverReq>) -> actix_web::Result<String> {
+    info!("register request for prover with name: {}", r.name);
+    if r.name == "" {
+        return Err(actix_web::error::ErrorBadRequest("empty name"));
+    }
+    let storage = match data.connection_pool.access_storage() {
+        Ok(s) => s,
+        Err(e) => return Err(actix_web::error::ErrorInternalServerError(e)),
+    };
+    let id = match storage.register_prover(&r.name) {
+        Ok(id) => id,
+        Err(e) => return Err(actix_web::error::ErrorInternalServerError(e)),
+    };
+    Ok(id.to_string())
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BlockToProveRes {
+    pub prover_run_id: i32,
+    pub block: i64,
+}
+
+fn block_to_prove(
+    data: web::Data<AppState>,
+    r: web::Json<ProverReq>,
+) -> actix_web::Result<HttpResponse> {
+    info!("request block to prove from worker: {}", r.name);
+    if r.name == "" {
+        return Err(actix_web::error::ErrorBadRequest("empty name"));
+    }
+    let storage = match data.connection_pool.access_storage() {
```

Comment:
see how it is done in the rest api. its just `data.access_storage()?`

Reply:
Is this the one that creates new connection pool internally and calls pool.access_storage on it? What is the point of connection pool?

Reply:
no, it does the same stuff at this one, but instead of repeating the same 3 lines of code everywhere it is just one simple expression. 
`let storage = data.access_storage()?`